### PR TITLE
[FLINK-20298][docs] Replace usage of {{ site.baseurl }} in documentation

### PR DESCRIPTION
**Pull request by Shawn-Hx**
## What is the purpose of the change

Replace usage of `{{ site.baseurl }}` in documentation.

## Brief change log

Change `{{ site.baseurl }}` to `{% link ... %}`.

## Verifying this change

Execute `docs/build_docs.sh` and `docs/check_links.sh` without errors.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by Shawn-Hx** [comment]
Hi, @dawidwys @sjwiesman 
Could you help to review at your convenience ?
Thanks~

**Review by dawidwys — CHANGES_REQUESTED** [review]
Thank you for the PR @Shawn-Hx 

Could you remove the `site.baseurl` from `.html` files as well? E.g. `_layouts/redirect.html`, `_layouts/base.html`?
A few tips from my brief experimentations and spotted problems:
* We need to change all  `redirect: /link/to/page/page.html` to `redirect: link/to/page/page.md`, otherwise the `{% link %}` notation won't work in `redirect.html`
* There are some redirected pages with broken links. You will see that when you change the `redirect.html` and try to build docs. There is one particulary ungrateful as it renders to `.md` which is hard to track back. The problematic file is `redirects/dependencies.md`. We do not have the `redirect: ` variable at all.
* We will need to implement a {%link_zh %} that will link to the Chinese version.
* Instead of `{{ site.baseurl }}{{ p.url }}` we should use `{% link {{ p.path }} %}`

**Comment by Shawn-Hx** [comment]
Thanks for the review @dawidwys 
I need some time to be familiar with the grammar of Jekyll.
I wonder if I could open another JIRA ticket to solve the remaining problems because this PR has changed a large number of files, and if others change these files there might be lots of conflicts to resolve.

**Comment by dawidwys** [comment]
@Shawn-Hx I think that is fair. Could you please create corresponding JIRA issues so that we do not forget about it. I will review the current PR in the meantime.

**Comment by Shawn-Hx** [comment]
Thanks @dawidwys 
I have created [FLINK-20422](https://issues.apache.org/jira/browse/FLINK-20422). Please take a look.

**Comment by Shawn-Hx** [comment]
@dawidwys I have resolved the conflicts, but I fount that `deployment/resource-providers/hadoop.md` don't exist so `build_docs.sh` will fail.

**Comment by dawidwys** [comment]
Yes, I've also stumbled upon this when trying to move your PR forward. I created the https://issues.apache.org/jira/browse/FLINK-20426 which should be fixed soon.

**Comment by Shawn-Hx** [comment]
@dawidwys 
I noticed that the problem has been fixed and resolved the conflicts. Now `./build_docs.sh` and `./check_links.sh` can be executed without error, PTAL :)

File: docs/dev/execution_configuration.md [hunk 1]
```diff
@@ -50,9 +50,9 @@ With the closure cleaner disabled, it might happen that an anonymous user functi
...
+- `getExecutionRetryDelay()` / `setExecutionRetryDelay(long executionRetryDelay)` Sets the delay in milliseconds that the system waits after a job has failed, before re-executing it. The delay starts after all tasks have been successfully stopped on the TaskManagers, and once the delay is past, the tasks are re-started. This parameter is useful to delay re-execution in order to let certain time-out related failures surface fully (like broken connections that have not fully timed out), before attempting a re-execution and immediately failing again due to the same problem. This parameter only has an effect if the number of execution re-tries is one or more. This is deprecated, use [restart strategies]({% link redirects/restart_strategies.md %}) instead.
```
Comment (dawidwys) [inline docs/dev/execution_configuration.md:55]: We should not link to redirects. Redirects are only for pages that do no longer exist.

**Review by dawidwys — APPROVED** [review]
Thank you for the work! I had one comment and there were some remaining occurrences after a rebase. I addressed those and merged.

**Pull request closed without merging**